TestGenerateOpus_InvalidModel_NotTriggered hangs up to 600s when `claude` is on PATH

## Summary

`internal/longmemeval/claude_additions_test.go:61` `TestGenerateOpus_InvalidModel_NotTriggered` invokes the **real `claude` CLI** on any machine where `claude` is on `PATH`, blocking the test for up to `generateTimeout = 600s` (`internal/longmemeval/claude.go:32`).

The test's comment assumes *"there is no real claude binary in CI"* — true in CI, **false on dev machines and the codex-poll runner** (`/home/psimmons/.local/bin/claude` is present). There it spawns `claude --print --model opus` (a real, billed Opus generation) and the `os/exec` → `io.Copy` read blocks until the 600s deadline.

## Impact

- `go test ./internal/longmemeval/ -count=1` hangs ~10 min locally, then FAILs the package on timeout.
- **This is what stalled the codex-poll run for #919** (2026-06-01): Codex completed its #919 implementation but its full-suite gate hung on this test, so it never committed/pushed. Poll labeled the issue `agent/codex/stalled`.
- Runs a real (billed) Opus generation as a side effect of `go test`.

## Reproduction

```
$ which claude            # /home/psimmons/.local/bin/claude
$ go test ./internal/longmemeval/ -run TestGenerateOpus_InvalidModel_NotTriggered -timeout 60s -v
panic: test timed out after 60s   # blocked in os/exec reading claude's stdout
```

## Root cause

The test passes `context.Background()` (no deadline). `runClaude` derives `context.WithTimeout(ctx, 600s)`, so the real `claude` invocation can run for the full 10 minutes.

## Suggested fix

Make the test deterministic regardless of whether `claude` is installed — pass a short-deadline context so it fails fast while still asserting the error is **not** the `ErrDisallowedModel` sentinel:

```go
ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
defer cancel()
_, err := longmemeval.GenerateOpus(ctx, "prompt", 0)
if err != nil && strings.Contains(err.Error(), "disallowed model") {
    t.Errorf("GenerateOpus should use an allowed model but got: %v", err)
}
```

This preserves the test's intent (opus is in the allowlist → never the disallowed-model error) without invoking a real, billed, 600s generation.

Found during manual inspection of the stalled #919 codex-poll run.

--- a/internal/longmemeval/claude_additions_test.go
+++ b/internal/longmemeval/claude_additions_test.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/petersimmons1972/engram/internal/longmemeval"
 )
@@ -62,7 +63,14 @@ func TestGenerateOpus_InvalidModel_NotTriggered(t *testing.T) {
 	// GenerateOpus passes "opus" which IS in the allowlist. The call will fail
 	// because there is no real claude binary in CI, but the error must NOT be
 	// the model-rejection sentinel.
-	ctx := context.Background()
+	//
+	// Use a 2s deadline so the test completes quickly on machines where a real
+	// `claude` binary is on PATH. The deadline does not affect what we're
+	// testing: the model-allowlist check in runClaude is synchronous and happens
+	// before any exec, so ErrDisallowedModel (if it were returned) arrives in
+	// well under 1 ms regardless of the deadline. (#954)
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
 	_, err := longmemeval.GenerateOpus(ctx, "prompt", 0)
 	if err != nil && strings.Contains(err.Error(), "disallowed model") {
 		t.Errorf("GenerateOpus should use an allowed model but got: %v", err)

